src/coreclr/inc/gcinfo.h

```diff
@@ -59,9 +60,18 @@ struct GCInfoToken
     PTR_VOID Info;
     UINT32 Version;
 
-    static UINT32 ReadyToRunVersionToGcInfoVersion(UINT32 readyToRunMajorVersion)
+    static UINT32 ReadyToRunVersionToGcInfoVersion(UINT32 readyToRunMajorVersion, UINT32 readyToRunMinorVersion)
     {
-        // GcInfo version is current from  ReadyToRun version 2.0
+        static_assert(MINIMUM_READYTORUN_MAJOR_VERSION == 9,
+            "when min version moves ahead, revisit the following logic, also check https://github.com/dotnet/runtime/issues/98871");
+
+        _ASSERT(readyToRunMajorVersion >= MINIMUM_READYTORUN_MAJOR_VERSION);
+
+        // R2R 9.0 and 9.1 use GCInfo v2
+        // R2R 9.2 uses GCInfo v3
+        if (readyToRunMajorVersion == 9 && readyToRunMinorVersion < 2)
+            return 2;
```

Comment:
```suggestion
// Delete once MINIMUM_READYTORUN_MAJOR_VERSION is bumped to 10+
#if MINIMUM_READYTORUN_MAJOR_VERSION < 10
        // R2R 9.0 and 9.1 use GCInfo v2
        // R2R 9.2 uses GCInfo v3
        if (readyToRunMajorVersion == 9 && readyToRunMinorVersion < 2)
            return 2;
#endif            
```
You can make the eventual cleanup more straightforward. We are tagging places that can use cleanup with min version bump with `MINIMUM_READYTORUN_MAJOR_VERSION`. https://github.com/dotnet/runtime/blob/99a79640e122436a74ccddf8e2c2337e43e831f1/src/coreclr/inc/readytorun.h#L20-L21 has comment about it.

Reply:
> If you update this, ensure you run `git grep MINIMUM_READYTORUN_MAJOR_VERSION`

I think a github issue to track the cleanup items would be useful. Not everything can be just tagged with `MINIMUM_READYTORUN_MAJOR_VERSION` in the code. It feels like we can put a lot more context in a tracking issue and keep adding as needed.

Also, I would not put the entire cleanup on the person who happens to rev the version. I think their duty should be just activate the tracking issue for the current milestone and notify the owners. - Perhaps split the tracking issue at that time into sub-issues, if cleanup is completely unrelated - i.e. suspension vs. SVE work.

Reply:
basically - whether it is a static assert or just a comment, I guess, makes little difference, but having an issue describing possible cleanups seems useful.

Reply:
I think the best way to prep for future cleanups is to make it very straightforward: Once X happens, delete Y and uncomment Z. Anybody can do that. It can be a comment in the code or issue on the side, I do not think that it matters a whole lot.

Reply:
I guess my actual concern is that there is a lot that can be cleaned up in the GCInfo/GCStress area once we rev the version and I am afraid to miss when the next opportunity opens for that.